### PR TITLE
Epic/changes according to oded review

### PR DESCRIPTION
**Pull request by alonshp**

### CODE REVIEW

File: FinalProject/DataModel/SphereCollectionDataSource.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
+//
+//  SphereCollectionDataSource.swift
+//  FinalProject
+//
+//  Created by Alon Shprung on 5/1/18.
+//  Copyright © 2018 Alon Shprung. All rights reserved.
+//
+
+import UIKit
+class SphereCollectionDataSource: NSObject,UICollectionViewDataSource {
```
Comment (oded-regev) [inline FinalProject/DataModel/SphereCollectionDataSource.swift]: Looks great

File: FinalProject/ViewControllers/CollectionViewController.swift [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    }
+    
+    func loadSphereData() {
+        showLoadingHUD()
+        networkUtils.fetchSphereData(offset: offset) { (sphereData) in
+            self.dataSource.items = sphereData.items
+            self.hideLoadingHUD()
+            self.collectionView.reloadData()
+        }
+        updateOffset()
```
Comment (oded-regev) [inline FinalProject/ViewControllers/CollectionViewController.swift]: this should be placed inside the "success closure"

File: FinalProject/ViewControllers/CollectionViewController.swift [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    }
+    
+    func getMoreData() {
+        networkUtils.fetchSphereData(offset: offset) { (sphereData) in
+            self.dataSource.items?.append(contentsOf: sphereData.items)
+            self.collectionView.reloadData()
+        }
+        updateOffset()
```
Comment (oded-regev) [inline FinalProject/ViewControllers/CollectionViewController.swift]: this should be placed inside the "success closure"

File: FinalProject/ViewControllers/TableViewController.swift [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    
+    func loadSphereData() {
+        showLoadingHUD()
+        networkUtils.fetchSphereData(offset: offset) { (sphereData) in
+            self.dataSource.items = sphereData.items
+            self.hideLoadingHUD()
+            self.tableView.reloadData()
+            self.showSpinnerAtTheEndOfTheData()
+        }
+        updateOffset()
```
Comment (oded-regev) [inline FinalProject/ViewControllers/TableViewController.swift]: this should be placed inside the "success closure"

File: FinalProject/ViewControllers/TableViewController.swift [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    
+    func getMoreData() {
+        networkUtils.fetchSphereData(offset: offset) { (sphereData) in
+            self.dataSource.items?.append(contentsOf: sphereData.items)
+            let indexPaths = ((self.dataSource.items?.count)! - 10 ..< (self.dataSource.items?.count)!)
+                .map { IndexPath(row: $0, section: 0) }
+            self.tableView.insertRows(at: indexPaths, with: UITableViewRowAnimation.fade)
+        }
+        updateOffset()
```
Comment (oded-regev) [inline FinalProject/ViewControllers/TableViewController.swift]: this should be placed inside the "success closure"

**Comment by alonshp** [comment]
Changes done

**Comment by oded-regev** [comment]
Great work!!

**Pull request merged**